app/Policies/CommentPolicy.php

```diff
@@ -26,7 +26,7 @@ public function manage(User $user): bool
 
     public function view(?User $user, Comment $comment): bool
     {
-        return true;
+        return $user->isNotBanned();
```

Comment:
In this case, is `$user` the currently logged in user, or is it the author of the comment?

I believe it's the currently logged in user. In which case, if it's a moderator or administrator, I'd expect them to be able to view any comment, even if the user is banned.

Reply:
In this situation, `$user` is the author of the comment

```
$results = $comments->filter(function ($nextComment) use ($policy) {
    return $policy->view($nextComment->user, $nextComment);
})->map(function ($nextComment) {
    return [
        'User' => $nextComment->user->username,
        'Submitted' => $nextComment->Submitted,
        'CommentText' => $nextComment->Payload,
    ];
});
```

Reply:
I think I've confused myself with this and may need help understanding the implementation details.

Is this checking if the comment author can see their own comment?

Reply:
Hah, yep. It seems I have also confused myself with this! It should be `return $policy->view(Auth::user(), $nextComment);`